src/lib.rs

```diff
@@ -742,27 +745,22 @@ impl RenderPass {
             queue.write_buffer(&buffer.buffer, 0, data);
         }
     }
-}
 
-impl epi::TextureAllocator for RenderPass {
-    fn alloc_srgba_premultiplied(
-        &mut self,
-        size: (usize, usize),
-        srgba_pixels: &[egui::Color32],
-    ) -> egui::TextureId {
-        let id = self.next_user_texture_id;
-        self.next_user_texture_id += 1;
+    // -------
+
+    /// Creates a texture at a certain user texture location
+    pub fn set_texture(&mut self, id: u64, image: epi::Image) -> egui::TextureId {
```

Comment:
Shouldn't the `id` be selected for the caller? It's an internal implementation detail. The actual id allocated is already included in the return value, which should be good enough? I can't think of any cases where I would want to specify the id when creating a texture.